### PR TITLE
Validator changes

### PR DESCRIPTION
**Pull request by strickinato**
If you ever want to validate your JSON and then use the values, right now that's impossible because the validator doesn't expose the validated values.

This PR provides a way to access them

### CODE REVIEW

File: src/JsonSchema/Validator.elm [hunk 1]
```diff
@@ -41,283 +41,371 @@ type ErrorMessage
...
+
+{-| This function validates your JSON against a schema.
+
+If validation succeeds, you can then gain access to the data you need
+
+If validation fails, you can extract a list of errors
+
+-}
+getValidatedValue : Pointer -> Schema -> Value -> ValidatedValue
+getValidatedValue pointer schema v =
```
Comment (jwoudenberg) [inline src/JsonSchema/Validator.elm:79]: 💭 Do you think we'd win anything by using [elm-verify](http://package.elm-lang.org/packages/stoeffel/elm-verify/latest) for this?

File: src/JsonSchema/Validator.elm [hunk 2]
```diff
@@ -41,283 +41,371 @@ type ErrorMessage
...
+type ValidatedValue
+    = Invalid (List Error)
+    | StringValue String
+    | ArrayValue (Array.Array ValidatedValue)
+    | ObjectValue (Dict String ValidatedValue)
+    | BoolValue Bool
+    | FloatValue Float
+    | IntValue Int
+    | LazyValue (List Error)
+    | Valid Value
```
Comment (jwoudenberg) [inline src/JsonSchema/Validator.elm:56]: 😻

File: src/JsonSchema/Validator.elm [hunk 3]
```diff
@@ -41,283 +41,371 @@ type ErrorMessage
...
-getDecodeError res =
-    case res of
-        Ok e ->
-            e
-
-        Err e ->
-            [ ( [], DecodeError e ) ]
-
+combineValidations : ValidatedValue -> List ValidatedValue -> ValidatedValue
+combineValidations success validations =
```
Comment (jwoudenberg) [inline src/JsonSchema/Validator.elm]: 😻

**Review by jwoudenberg — APPROVED** [review]
This looks great Strick, thank you!

One of you documentation comments made me think of elm-verify which was built around the idea of a validation either producing errors or a 'validated value'. I'm not sure if it will add much value here though, so I'm also happy to merge this as is!

**Comment by strickinato** [comment]
@jwoudenberg ---

I really should have merged when I had the change :( 

I ended up having a slightly tricky rebase since Tuples got involved

It's pretty weird because the idea of having a tuple of unknown length isn't really possible in elm, so I ended up having to model that value as a tuple of `(Int, ArrayValue)

**Comment by strickinato** [comment]
I actually had to change the way this all works anyway for my form - so I'm closing in favor of some other work I'll merge together when that's ready

**Comment by jwoudenberg** [comment]
You're really building up anticipation Strick, excited to see more of the form editor 😀.

**Pull request closed without merging**